pex/pex.py

```diff
@@ -617,10 +617,22 @@ def execute_script(self, script_name):
         dist_script = get_script_from_distributions(script_name, dists)
         if not dist_script:
             return "Could not find script {!r} in pex!".format(script_name)
-        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist))
-        return self.execute_content(
-            dist_script.path, dist_script.read_contents(), argv0=script_name
-        )
+
+        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist_script.dist))
+        if dist_script.is_python_script():
+            return self.execute_content(
+                dist_script.path, dist_script.read_contents(), argv0=script_name
+            )
+        else:
+            return self.execute_external(dist_script.path)
+
+    def execute_external(self, cmd):
```

Comment:
This can be a staticmethod ~like `execute_content` below since it does not use self or cls.